attic/modules/potentials/bonded.py

```diff
@@ -0,0 +1,54 @@
+import jax.numpy as jnp
+
+def restraint(conf, lamb, params, lamb_flags, box, bond_idxs):
+    r"""
+    Compute the harmonic bond energy given a collection of molecules.
+
+    This implements a harmonic bond potential:
+        V(conf; lamb) = \sum_bond kbs[bond] term[bond]^2
+
+        where term[bond] = 1 - exp(-a0s[bond]*(dij[bond] - b0s[bond]))
+        and where
+            dij[bond] = distance[bond] + f_lambda^2
+            and where
+                f_lambda = lamb * lamb_flags
```

Comment:
Main motivation for removal: This dependence on lamb and lamb_flags is a bit different from that of other `bonded` potentials.

(Changing the signatures of the other `bonded` potentials will be marginally simpler if this potential doesn't have to be updated also.)